bigtable/beam/batch-write-flow-control-example/src/main/java/bigtable/BatchWriteFlowControlExample.java

```diff
@@ -0,0 +1,228 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bigtable;
+
+import com.google.bigtable.v2.Mutation;
+import com.google.bigtable.v2.Mutation.SetCell;
+import com.google.cloud.bigtable.beam.CloudBigtableIO;
+import com.google.cloud.bigtable.beam.CloudBigtableTableConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.gcp.bigtable.BigtableIO;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.util.Bytes;
+
+/*
+An example pipeline to demo the batch write flow control feature.
+ */
+public class BatchWriteFlowControlExample {
+
+  static long numRows;
+
+  static final String COLUMN_FAMILY = "cf";
+  static final SecureRandom random = new SecureRandom();
+
+  public static void main(String[] args) {
+    BigtablePipelineOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(BigtablePipelineOptions.class);
+
+    Preconditions.checkNotNull(options.getBigtableInstanceId());
+
+    numRows = options.getBigtableRows();
+
+    generateData(options);
+  }
+
+  static void generateData(BigtablePipelineOptions options) {
+
+    System.out.println("Generating " + options.getBigtableRows() + " rows, each "
+        + options.getBigtableColsPerRow() + " columns, " + options.getBigtableBytesPerCol()
+        + " bytes per column, " + options.getBigtableColsPerRow() * options.getBigtableBytesPerCol()
+        + " bytes per row.");
+
+    String generateLabel = String
+        .format("Generate %d rows for table %s", numRows, options.getBigtableTableId());
+    String mutationLabel = String
+        .format("Create mutations that write %d columns of total %d bytes to each row",
+            options.getBigtableColsPerRow(), options.getBigtableColsPerRow()
+                * options.getBigtableBytesPerCol());
+
+    Pipeline p = Pipeline.create(options);
+
+    PCollection<Long> numbers = p.apply(generateLabel, GenerateSequence.from(0).to(numRows));
+
+    if (options.getUseCloudBigtableIo()) {
```

Comment:
do we need the code showing both ways for this? I know we recommend customers use one of these over the other but can never remember which one. Ideally we just show the recommended way. Or if the goal is to just show how to activate the flow control on the client then we could keep both but maybe the data being written can be simplified to not have as much information for customers to take in. 

For example, we could just write one row using each client and that can reduce the need to generate sequences and all the additional helper functions.

Reply:
Yes we recommend BigtableIO bug we need examples to show both connector as they're configured differently.

I'm hoping this example can demo the effect of flow control so that the job needs to write sufficient amount of data. Also user can play with the data shape with the parameters to understand flow control more if they want.